OptimizelySDK/Optimizely.cs

```diff
@@ -529,7 +586,14 @@ public virtual bool IsFeatureEnabled(string featureKey, string userId, UserAttri
         /// <returns>bool | Feature variable value or null</returns>
         public bool? GetFeatureVariableBoolean(string featureKey, string variableKey, string userId, UserAttributes userAttributes = null)
         {
-            return GetFeatureVariableValueForType<bool?>(featureKey, variableKey, userId, userAttributes, FeatureVariable.VariableType.BOOLEAN);
+            var config = ProjectConfigManager?.GetConfig();
+
+            if (!IsValid && config == null) {
+                Logger.Log(LogLevel.ERROR, "Datafile has invalid format. Failing 'activate'.");
+                return null;
+            }
+
+            return GetFeatureVariableValueForType<bool?>(featureKey, variableKey, userId, userAttributes, FeatureVariable.VariableType.BOOLEAN, config);
```

Comment:
Why do we need to pass the config to it? Why not have the `GetFeatureVariableValueForType` method use the `ProjectConfigManager`?